src/Microsoft.OData.Edm/EdmModelVisitor.cs

```diff
@@ -193,11 +261,77 @@ public void VisitExpression(IEdmExpression expression)
+                    return this.ProcessFloatingConstantExpressionAsync((IEdmFloatingConstantExpression)expression);
+                case EdmExpressionKind.FunctionApplication:
+                    return this.ProcessFunctionApplicationExpressionAsync((IEdmApplyExpression)expression);
+                case EdmExpressionKind.GuidConstant:
+                    return this.ProcessGuidConstantExpressionAsync((IEdmGuidConstantExpression)expression);
+                case EdmExpressionKind.If:
+                    return this.ProcessIfExpressionAsync((IEdmIfExpression)expression);
+                case EdmExpressionKind.IntegerConstant:
+                    return this.ProcessIntegerConstantExpressionAsync((IEdmIntegerConstantExpression)expression);
+                case EdmExpressionKind.IsOf:
+                    return this.ProcessIsOfExpressionAsync((IEdmIsOfExpression)expression);
+                case EdmExpressionKind.LabeledExpressionReference:
+                    return this.ProcessLabeledExpressionReferenceExpressionAsync((IEdmLabeledExpressionReferenceExpression)expression);
+                case EdmExpressionKind.Labeled:
+                    return this.ProcessLabeledExpressionAsync((IEdmLabeledExpression)expression);
+                case EdmExpressionKind.Null:
+                    return this.ProcessNullConstantExpressionAsync((IEdmNullExpression)expression);
+                case EdmExpressionKind.Path:
+                    return this.ProcessPathExpressionAsync((IEdmPathExpression)expression);
+                case EdmExpressionKind.PropertyPath:
+                    return this.ProcessPropertyPathExpressionAsync((IEdmPathExpression)expression);
+                case EdmExpressionKind.NavigationPropertyPath:
+                    return this.ProcessNavigationPropertyPathExpressionAsync((IEdmPathExpression)expression);
+                case EdmExpressionKind.AnnotationPath:
+                    return this.ProcessAnnotationPathExpressionAsync((IEdmPathExpression)expression);
+                case EdmExpressionKind.Record:
+                    return this.ProcessRecordExpressionAsync((IEdmRecordExpression)expression);
+                case EdmExpressionKind.StringConstant:
+                    return this.ProcessStringConstantExpressionAsync((IEdmStringConstantExpression)expression);
+                case EdmExpressionKind.TimeOfDayConstant:
+                    return this.ProcessTimeOfDayConstantExpressionAsync((IEdmTimeOfDayConstantExpression)expression);
+                case EdmExpressionKind.DurationConstant:
+                    return this.ProcessDurationConstantExpressionAsync((IEdmDurationConstantExpression)expression);
+                case EdmExpressionKind.None:
+                    return this.ProcessExpressionAsync(expression);
+                default:
+                    throw new InvalidOperationException(Edm.Strings.UnknownEnumVal_ExpressionKind(expression.ExpressionKind));

```

Comment:
Is it possible to return a faulted task instead? If this exception was thrown, it would be caught at the method up in the call stack that is defined with the `async` keyword. I don't know if it's possible to cause this block to be hit. If this is unreachable code, it'd make more sense to use a `Contract.Assert` instead. If the code is reachable, write a test for that scenario.

Reply:
@habbes @gathogojr Looking at the switch expression and the cases, the `default` will be unreachable here unless another expressionKind is added.
Things I can do here:
1. Use `async/await` instead of directly returning Task. 
2. Use `Contract.Assert` for debugging purposes.
3. Return faulted task using Task.FromException. For example:
`return Task.FromException(new InvalidOperationException(Edm.Strings.UnknownEnumVal_TypeKind(definition.TypeKind)));`

Reply:
@habbes @gathogojr I have opted to use `Contract.Assert` and return faulted task:

```cs
...
 Contract.Assert(false, Edm.Strings.UnknownEnumVal_ExpressionKind(expression.ExpressionKind));
 return Task.FromException<InvalidOperationException>(new InvalidOperationException(Edm.Strings.UnknownEnumVal_ExpressionKind(expression.ExpressionKind)));
...
```